### PR TITLE
881 no --drop-state flag for first run

### LINKED ISSUE #881: Cannot launch from scratch without --drop-state flag

### PR DESCRIPTION
**Pull request by kuvaldini**
Anyway database initialization required:
```
initdb ~/iroha/noda3_db/
postgres -D ~/iroha/noda3_db/ -d1 -p5432 &
createuser -s iroha_user -p5432
createdb iroha_user -p5432 -Oiroha_user
```

Closes #881

### CODE REVIEW

File: irohad/subscription/subscription_manager.hpp [hunk 1]
```diff
@@ -16,6 +16,17 @@
...
+namespace std {
+
+  template <typename To, typename From>
+  inline std::shared_ptr<To> reinterpret_pointer_cast(std::shared_ptr<From> const & ptr) noexcept {
+    return std::shared_ptr<To>(ptr, reinterpret_cast<To *>(ptr.get()));
+  }
+
+}
+#endif
+
```
[the remark this answers is not in the document]
Comment (iceseer) [reply irohad/subscription/subscription_manager.hpp:29]: Will add in additional PR.

**Review by iceseer — CHANGES_REQUESTED** [review]

File: irohad/main/impl/pg_connection_init.cpp [hunk 2]
```diff
@@ -151,7 +151,18 @@ PgConnectionInit::prepareWorkingDatabase(
     StartupWsvDataPolicy startup_wsv_data_policy,
     const PostgresOptions &options) {
   return getMaintenanceSession(options) | [&](auto maintenance_sql) {
-    if (startup_wsv_data_policy == StartupWsvDataPolicy::kReuse) {
+    int work_db_exists;
```
Comment (kuvaldini) [inline irohad/main/impl/pg_connection_init.cpp:154]: It should be bool but soci does not accept bool. int works as logical boolean here

**Review by kamilsa — APPROVED** [review]

File: irohad/subscription/subscription_manager.hpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (kuvaldini) [reply irohad/subscription/subscription_manager.hpp]: removed.

**Review by iceseer — APPROVED** [review]

**Comment by lebdron** [comment]
@iceseer @kuvaldini did you check that tests don't pass?

**Comment by kuvaldini** [comment]
Оо нет

**Pull request merged**